scripts/src/java/org/oppia/android/scripts/coverage/RunCoverage.kt

```diff
@@ -102,30 +106,100 @@ class RunCoverage(
    * @return a list of lists containing coverage data for each requested test target, if
    *     the file is exempted from having a test file, an empty list is returned
    */
-  fun execute() {
+  fun execute() = runBlocking {
     val testFileExemptionList = loadTestFileExemptionsProto(testFileExemptionTextProto)
       .testFileExemptionList
       .filter { it.testFileNotRequired }
       .map { it.exemptedFilePath }
 
     if (filePath in testFileExemptionList) {
       println("This file is exempted from having a test file; skipping coverage check.")
-      return
-    }
+    } else {
+      val testFilePaths = findTestFile(repoRoot, filePath)
+      if (testFilePaths.isEmpty()) {
+        error("No appropriate test file found for $filePath")
+      }
 
-    val testFilePaths = findTestFile(repoRoot, filePath)
-    if (testFilePaths.isEmpty()) {
-      error("No appropriate test file found for $filePath")
-    }
+      val testTargets = bazelClient.retrieveBazelTargets(testFilePaths)
 
-    val testTargets = bazelClient.retrieveBazelTargets(testFilePaths)
+      /*since I couldn't actually find any multi test target : file ones to test
+    * I am for now introducing mock data to test multi aggregated coverage report
+    * also that's going to save me a light year :|
+    * */
 
-    val coverageReports = testTargets.mapNotNull { testTarget ->
-      runCoverageForTarget(testTarget)
-    }
+      val deferredCoverageReports = testTargets.map { testTarget ->
+        runCoverageForTarget(testTarget)
+      }
+
+      val coverageReports = deferredCoverageReports.awaitAll()
 
-    coverageReports.takeIf { it.isNotEmpty() }?.run {
-      val reporter = CoverageReporter(repoRoot, this, reportFormat)
+      // For reference. (to be removed)
```

Comment:
Here, for review purposes, I think it is enough to add an explanation comment in a few lines than the code block.

Reply:
Got it. To still have it for reference I moved them to the PR comments - https://github.com/oppia/oppia-android/pull/5459#issuecomment-2253028530